test/graph_test.cpp

```diff
@@ -1,26 +1,88 @@
 #include <gtest/gtest.h>
+#include <fstream>
 #include "../include/graph.h"
+#include "util/deterministic.h"
+#include "util/graph_gen.h"
 
 TEST(GraphTestSuite, SmallGraphConnectivity) {
-  unsigned long long int num_nodes = 1000;
+  const std::string fname = __FILE__;
+  size_t pos = fname.find_last_of("\\/");
+  const std::string curr_dir = (std::string::npos == pos) ? "" : fname.substr(0, pos);
+  ifstream in{curr_dir + "/res/multiples_graph_1024.txt"};
+  Node num_nodes;
+  in >> num_nodes;
+  long m;
+  in >> m;
+  Node a, b;
   Graph g{num_nodes};
-  for (unsigned i=1;i<num_nodes;++i) {
-    for (unsigned j = i*2;j<num_nodes;j+=i) {
-      g.update({{i,j}, INSERT});
-    }
+  while (m--) {
+    in >> a >> b;
+    g.update({{a, b}, INSERT});
   }
-  ASSERT_EQ(2, g.connected_components().size());
+  g.set_cum_in(curr_dir + "/res/multiples_graph_1024.txt");
+  ASSERT_EQ(78, g.connected_components().size());
 }
 
 TEST(GraphTestSuite, IFconnectedComponentsAlgRunTHENupdateLocked) {
-  unsigned long long int num_nodes = 1000;
+  const std::string fname = __FILE__;
+  size_t pos = fname.find_last_of("\\/");
+  const std::string curr_dir = (std::string::npos == pos) ? "" : fname.substr(0, pos);
+  ifstream in{curr_dir + "/res/multiples_graph_1024.txt"};
+  Node num_nodes;
+  in >> num_nodes;
+  long m;
+  in >> m;
+  Node a, b;
   Graph g{num_nodes};
-  for (unsigned i=1;i<num_nodes;++i) {
-    for (unsigned j = i*2;j<num_nodes;j+=i) {
-      g.update({{i,j}, INSERT});
-    }
+  while (m--) {
+    in >> a >> b;
+    g.update({{a, b}, INSERT});
   }
+  g.set_cum_in(curr_dir + "/res/multiples_graph_1024.txt");
   g.connected_components();
   ASSERT_THROW(g.update({{1,2}, INSERT}), UpdateLockedException);
   ASSERT_THROW(g.update({{1,2}, DELETE}), UpdateLockedException);
 }
+
+TEST(GraphTestSuite, TestRandomGraphGeneration) {
+  generate_stream();
+  GraphVerifier graphVerifier {"./cum_sample.txt"};
```

Comment:
I'm not 100% sure about this syntax but isn't cum_sample a default arg. Do we need to specify.

Reply:
You're right, it's not needed